src/platform/silabs/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -159,6 +161,32 @@ CHIP_ERROR SlWiFiDriver::ConnectWiFiNetwork(const char * ssid, uint8_t ssidLen,
     return CHIP_NO_ERROR;
 }
 
+// TODO: Re-write implementation with proper driver based callback
+void SlWiFiDriver::UpdateNetworkingStatus()
+{
+    if (mpStatusChangeCallback == nullptr)
+    {
+        ChipLogError(NetworkProvisioning, "networkStatusChangeCallback is nil");
+        return;
+    }
+
+    if (mStagingNetwork.ssidLen == 0)
+    {
+        ChipLogError(NetworkProvisioning, "ssidLen is 0");
+        return;
+    }
+
+    ByteSpan networkId = ByteSpan((const unsigned char *) mStagingNetwork.ssid, mStagingNetwork.ssidLen);
```

Comment:
This should probably use `Uint8::from_char`.  But I do wonder why this is storing `char *` and not `uint8_t *` to start with.

Reply:
`mStagingNetwork.ssid` holds a string and is a char array. Why should it be a uint8_t * ? Maybe a reinterpret_cast<const char *> should've used? Where are you casting for the const?